File: swift/raindrops/Sources/Raindrops/Raindrops.swift
//Solution goes in Sources
class Raindrops {

    let pairs: [(Int, String)] = [ (3, "Pling"), (5, "Plang"), (7, "Plong")]
    let num: Int
    
    init(_ num: Int) {
        self.num = num
    }
    
    var sounds: String {
        get {
            var sound = ""
            pairs.map{(f, s) in 
                sound += self.num % f == 0 ? s : ""
            }
            return sound == "" ? String(num) : sound
        }
    }
}

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

func Convert(num int) string {
	result := ""

	if num%3 == 0 {
		result += "Pling"
	}

	if num%5 == 0 {
		result += "Plang"
	}

	if num%7 == 0 {
		result += "Plong"
	}

	if result == "" {
		result = strconv.Itoa(num)
	}

	return result
}
